assets/src/components/Lightbox/setupLightboxForImageBlocks.js

```diff
@@ -0,0 +1,34 @@
+import { Lightbox } from './Lightbox';
+
+export const setupLightboxForImageBlocks = function() {
+    // We can't use `createPortal` outside `render()`,
+    // `React.render()` needs a node, even if it ends up being empty.
+    // See: https://github.com/facebook/react/issues/12653#issuecomment-382851495
+    const lightBoxNode = document.createElement('div');
+    lightBoxNode.style.position = 'fixed';
+  
+    document.body.appendChild(lightBoxNode);
+  
+    const imageBlocks = document.querySelectorAll('.wp-block-image:not(.force-no-lightbox)');
+    imageBlocks.forEach((imageBlock, index) => {   
+      const image = imageBlock.querySelector('img');
+  
+      const item = {
+        src: image.src ? image.src : image.dataset.src,
+        w: 0,
+        h: 0
+      };
+    
+      const caption = imageBlock.querySelector('figcaption');
+      if (caption) {
+        item.title = caption.innerHTML;
+      }
+  
+      imageBlock.querySelector('img').addEventListener('click', () => wp.element.render(
+          <Lightbox isOpen={true} index={index} items={[item]} onClose={() => {}} />,
```

Comment:
You could also set onClose defaulting to a noop in the components arguments, then you could exclude `onClose` prop here.
```js
export const Lightbox = ({ index, isOpen, items, onClose = () => {} }) => {
...